components/ringcentral/ringcentral.app.mjs

```diff
@@ -0,0 +1,229 @@
+          value,
+        }));
+      },
+    },
+  },
+  methods: {
+    _authToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      const platform = this.$auth.instancetype ?? "https://platform";
+      return `${platform}.ringcentral.com/restapi/v1.0`;
+    },
+    getHeaders() {
+      const authToken = this._authToken();
+      return {
+        "Authorization": `Bearer ${authToken}`,
+        "User-Agent": "@PipedreamHQ/pipedream v0.1",
+      };
+    },
+    async makeRequest({
+      $ = this, url, path, method = "get", ...args
+    } = {}) {
+      const config = {
+        method,
+        url: url ?? `${this._apiUrl()}${path}`,
+        headers: this.getHeaders(),
+        ...args,
+      };
+      try {
+        return await axios($, config);
+      } catch (error) {
+        throw error.response?.data?.message ?? error;
+      }
+    },
+    // https://developers.ringcentral.com/api-reference/Extensions/listExtensions
+    async _getExtensionList(args = {}) {
+      return this.makeRequest({
+        path: "/account/~/extension",
+        ...args,
+      });
+    },
+    // https://developers.ringcentral.com/api-reference/Devices/listExtensionDevices
+    async _getDeviceList({
+      extensionId = "~", ...args
+    }) {
+      return this.makeRequest({
+        path: `/account/~/extension/${extensionId}/device`,
+        ...args,
+      });
+    },
+    async getCountryList(args = {}) {
+      return this.makeRequest({
+        path: "/dictionary/country",
+        ...args,
+      });
+    },
+    // https://developers.ringcentral.com/api-reference/Call-Log/readUserCallLog
+    async getUserCallLogRecords({
+      accountId = "~", extensionId = "~", ...args
+    }) {
+      return this.makeRequest({
+        path: `/account/${accountId}/extension/${extensionId}/call-log`,
+        ...args,
+      });
+    },
+    async makeCallOut({
+      accountId = "~", ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/account/${accountId}/telephony/call-out`,
+        ...args,
+      });
+    },
+    async createMeeting({
+      accountId = "~", extensionId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/account/${accountId}/extension/${extensionId}/meeting`,
+        ...args,
+      });
+    },
+    async sendSMS({
+      accountId = "~", extensionId, ...args
+    }) {
+      return this.makeRequest({
+        method: "post",
+        path: `/account/${accountId}/extension/${extensionId}/sms`,
+        ...args,
+      });
+    },
+    // Details about the different webhook parameters can be found in the
+    // RingCentral API docs:
+    // https://developers.ringcentral.com/api-reference/Subscriptions/createSubscription
+    async createHook(args = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: "/subscription",
+        ...args,
+      });
+    },
+    deleteHook(hookId) {
```

Comment:
Is good wait for `deleteHook`, because if have an error will return to user interface.

You can check and fix this for the others sources too.